File: src/main/java/me/zoarial/NetworkArbiter/annotations/ZoarialObjectElement.java
package me.zoarial.NetworkArbiter.annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.FIELD)
public @interface ZoarialObjectElement {
    boolean optional() default false;
    int placement();
}

File: src/main/java/me/zoarial/networkArbiter/annotations/ZoarialObjectElement.kt
package me.zoarial.networkArbiter.annotations

import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

@Retention(RetentionPolicy.RUNTIME)
@Target(AnnotationTarget.FIELD)
annotation class ZoarialObjectElement(val optional: Boolean = false, val placement: Int)